presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -508,21 +508,45 @@ public PlanWithProperties visitTableWriter(TableWriterNode originalTableWriterNo
             PlanWithProperties tableWriter;
 
             if (!originalTableWriterNode.getTablePartitioningScheme().isPresent()) {
-                tableWriter = planAndEnforceChildren(
-                        new TableWriterNode(
-                                originalTableWriterNode.getId(),
-                                originalTableWriterNode.getSource(),
-                                originalTableWriterNode.getTarget(),
-                                variableAllocator.newVariable("partialrowcount", BIGINT),
-                                variableAllocator.newVariable("partialfragments", VARBINARY),
-                                variableAllocator.newVariable("partialcontext", VARBINARY),
-                                originalTableWriterNode.getColumns(),
-                                originalTableWriterNode.getColumnNames(),
-                                originalTableWriterNode.getTablePartitioningScheme(),
-                                originalTableWriterNode.getPreferredShufflePartitioningScheme(),
-                                statisticAggregations.map(StatisticAggregations.Parts::getPartialAggregation)),
-                        fixedParallelism(),
-                        fixedParallelism());
+                int taskWriterCount = getTaskWriterCount(session);
+                int taskConcurrency = getTaskConcurrency(session);
+                if (taskWriterCount == taskConcurrency) {
```

Comment:
can this be `if (taskWriterCount > taskConcurrency)` ?

Reply:
No. We want to allow `taskWriterCount ` and `taskConcurrency ` to be set independently. It should be allowed to set higher `taskWriterCount ` than `taskConcurrency `.